src/templating/native/nativeTemplateEngine.js

```diff
@@ -10,7 +13,16 @@ ko.nativeTemplateEngine.prototype['renderTemplateSource'] = function (templateSo
         templateNodes = templateNodesFunc ? templateSource['nodes']() : null;
 
     if (templateNodes) {
-        return ko.utils.makeArray(templateNodes.cloneNode(true).childNodes);
+        var nodes;
+        if (templateNodes instanceof Array) {
+            nodes = [];
+            for (var i = 0; i < templateNodes.length; ++i) {
```

Comment:
Why not `ko.utils.arrayForEach`?  If doing the loop here this should cache `templateNodes.length`